simrupt.c

```diff
@@ -9,6 +9,9 @@
 #include <linux/slab.h>
 #include <linux/version.h>
 #include <linux/workqueue.h>
+#if defined(CONFIG_X86)
+#include <linux/vmalloc.h>
+#endif
```

Comment:
I understand from the previous discussion that this compilation error may only occur on x86. However, it seems odd to wrap `#include <linux/vmalloc.h>` with `#if defined(CONFIG_X86)`, as `vmalloc` shouldn't be architecture-specific. Would it make more sense to remove the `#if defined(CONFIG_X86)`?